instruqt-tracks/network-infrastructure-automation/8-install-consul-terraform-sync/solve-workstation

```diff
@@ -8,6 +8,33 @@ echo ">>"
 cd /root/terraform/consul-tf-sync
 terraform apply -auto-approve > /root/terraform/consul-tf-sync/terraform.out
 
-sleep 90
+# Wait until things are actually set up
+n=0
+until [ $n -ge 20 ]; do
+    echo "Try ${n}"
+    rg=$(terraform output -state /root/terraform/vnet/terraform.tfstate resource_group_name)
+    echo "Checking consul-terraform-sync instance in resource group $rg:"
+    status=$(az vm get-instance-view -g $rg --name consul-terraform-sync | jq -r .provisioningState)
+    echo $status
+
+    if [ "${status}" != "Creating" ] && [ "${status}" != "Updating" ] && [ "${status}" != "Succeeded" ] ; then
+      echo "consul-terraform-sync instance is not provisioned. Wait a few moments and try again. Current state is: ${status}"
```

Comment:
The user won't actually need to try again during this loop, right?  It'll just loop 20 times and give up -- might be worth rephrasing so folks don't get confused or prematurely kill the apply.

Reply:
This is in the solve script, which only runs when skipping, or when running through the CI/CD testing. And it will only show up in the log. That said, I did just copy it in from the check script, so the wording may be a bit off, but end users won't see this in any place where they can interrupt the retries.

Reply:
I can change the wording easily enough, no reason to log that.